drivers/bmx280/bmx280.c

```diff
@@ -23,93 +25,322 @@
+    if (_read_burst(dev, DATA_BASE, RAW_DATA, BMX280_RAW_LEN) != BMX280_OK) {
+        goto err;
+    }
+
+    /* we are done reading from the device, so release the bus again */
+    _release(dev);
+    return BMX280_OK;
+
+err:
+    _release(dev);
+    return BMX280_ERR_BUS;
+}
 
-int bmx280_init(bmx280_t* dev, const bmx280_params_t* params)
+int bmx280_init(bmx280_t *dev, const bmx280_params_t *params)
 {
-    uint8_t chip_id;
+    assert(dev && params);
 
     dev->params = *params;
+    uint8_t reg;
 
-    /* Read chip ID */
-    chip_id = read_u8_reg(dev, BMX280_CHIP_ID_REG);
-    if ((chip_id != BME280_CHIP_ID) && (chip_id != BMP280_CHIP_ID)) {
-        DEBUG("[Error] Did not detect a BMX280 at address %02x (%02x != %02x or %02x)\n",
-              dev->params.i2c_addr, chip_id, BME280_CHIP_ID, BMP280_CHIP_ID);
+
+#ifdef BMX280_USE_SPI
+    /* configure the chip-select pin */
+    if (spi_init_cs(BUS, CS) != SPI_OK) {
+        DEBUG("[bmx280] error: unable to configure chip the select pin\n");
+        return BMX280_ERR_BUS;
+    }
+#endif
+
+    /* acquire bus bus, this also tests the bus parameters in SPI mode */
+    if (_acquire(dev) != BMX280_OK) {
+        DEBUG("[bmx280] error: unable to acquire bus\n");
+        return BMX280_ERR_BUS;
+    }
+
+    /* test the connection to the device by reading and verifying its chip ID */
+    if (_read_reg(dev, BMX280_CHIP_ID_REG, &reg) != BMX280_OK) {
+        DEBUG("[bmx280] error: unable to read chip ID from device\n");
         return BMX280_ERR_NODEV;
     }
+    if (reg != BMX280_CHIP_ID_VAL) {
+        DEBUG("[bmx280] error: invalid chip ID (0x%02x)\n", (int)reg);
+        _release(dev);
+        return BMX280_ERR_NODEV;
+    }
+
+    /* trigger a power-on reset sequence to reset all registers */
+    if (_write_reg(dev, BMEX80_RST_REG, RESET_WORD) != BMX280_OK) {
+        goto err;
+    }
+    /* wait for reset sequence to finish */
+    do {
+        if (_read_reg(dev, BMX280_STAT_REG, &reg) != BMX280_OK) {
+            goto err;
+        }
+    } while (reg != 0);
 
-    /* Read compensation data, 0x88..0x9F, 0xA1, 0xE1..0xE7 */
-    if (read_calibration_data(dev)) {
-        DEBUG("[Error] Could not read calibration data\n");
-        return BMX280_ERR_NOCAL;
+    /* read the compensation data from the sensor's ROM */
+    if (_read_calibration_data(dev) != BMX280_OK) {
+        DEBUG("[bmx280] error: could not read calibration data\n");
+        goto err;
     }
 
+    /* write basic device configuration: t_sb and filter values */
+    reg = (dev->params.t_sb | dev->params.filter);
+    if (_write_reg(dev, BMX280_CONFIG_REG, reg) != BMX280_OK) {
+        goto err;
+    }
+
+#if defined(MODULE_BME280_SPI) || defined(MODULE_BME280_I2C)
+    /* ctrl_hum must be written before ctrl_meas for changes to become
+     * effective */
+    reg = dev->params.humid_oversample;
+    if (_write_reg(dev, BME280_CTRL_HUM_REG, reg) != BMX280_OK) {
+        goto err;
+    }
+#endif
```

Comment:
It is not even possible to have a BMP280 and a BME280 using the same interface.

Reply:
Nope, it is not. BUT
a) it is an extremely rare use case to have both connected concurrently 
b) you might just use the `bme280` driver config with both devices as a workaround
c) the convention in RIOT we always used to follow is not to over complicate things for the last 1% of exotic use cases
d) **this has nothing to do with the content of this PR**